terraform/gitlab/gitlab.tf.json.template.py

```diff
@@ -742,6 +769,46 @@ def remove_inconsequential_statements(statements: List[JSON]) -> List[JSON]:
                 ]
             }
         },
+        "aws_s3_bucket": {
+            "gitlab": {
+                "bucket": f"edu-ucsc-gi-singlecell-azul-gitlab-{config.deployment_stage}-{aws.region_name}"
+            }
+        },
+        "aws_s3_bucket_policy": {
+            "gitlab": {
+                "bucket": "${aws_s3_bucket.gitlab.id}",
+                # FIXME:  Bucket lacks lifecycle policy
+                #         https://github.com/DataBiosphere/azul/issues/3620
+                "policy": json.dumps({
+                    "Version": "2012-10-17",
+                    "Statement": [
+                        {
+                            "Effect": "Allow",
+                            "Principal": {
+                                "AWS": f"arn:aws:iam::{elb_region_account_id[aws.region_name]}:root"
+                            },
+                            "Action": "s3:PutObject",
+                            "Resource": f"${{aws_s3_bucket.gitlab.arn}}/logs/alb/AWSLogs/{aws.account}/*"
+                        },
+                        *({
+                            "Effect": "Allow",
+                            "Principal": {
+                                "Service": "delivery.logs.amazonaws.com"
+                            },
+                            "Action": f"s3:{action}",
+                            "Resource": resource,
+                            **({"Condition": {
+                                "StringEquals": {
+                                    "s3:x-amz-acl": "bucket-owner-full-control"
+                                }
+                            }} if action == "PutObject" else {})
+                        } for action, resource in (("PutObject",
```

Comment:
https://github.com/DataBiosphere/azul/blob/develop/CONTRIBUTING.rst#14collection-literals

and F